### ISSUE TITLE
Create one blobfuse PVC will sometimes lead to two containers created finally

### ISSUE THREAD

**Issue by andyzhangx** [csi-driver/blobfuse-csi-driver#34]
**What happened**:
With below blobfuse CSI storage class:
```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: blobfuse.csi.azure.com
provisioner: blobfuse.csi.azure.com
parameters:
  skuName: Standard_LRS  #available values: Standard_LRS, Standard_GRS, Standard_RAGRS
reclaimPolicy: Retain #If set as "Delete" container would be removed after pvc deletion
volumeBindingMode: Immediate
```

Per below logs, the first `CreateVolume` request costs about 18s(including creating a new storage account), that would lead to another CreateVolume request.
```
I0423 07:20:45.195925       1 utils.go:106] GRPC call: /csi.v1.Controller/CreateVolume
I0423 07:20:45.195942       1 utils.go:107] GRPC request: name:"pvc-4a92e455-6598-11e9-9989-0e3cd97702b7" capacity_range:<required_bytes:10737418240 > volume_capabilities:<mount:<fs_type:"ext4" mount_flags:"--file-cache-timeout-in-seconds=120" mount_flags:"--use-https=true" > access_mode:<mode:MULTI_NODE_MULTI_WRITER > > parameters:<key:"skuName" value:"Standard_LRS" >
I0423 07:20:45.606774       1 azure_storageaccount.go:121] azure - no matching account found, begin to create a new account fuse4f945a18659811e99ef in resource group mc_andy-aks1135_andy-aks1135_eastus2, location: eastus2, accountType: Standard_LRS, accountKind: StorageV2
I0423 07:21:01.625287       1 utils.go:106] GRPC call: /csi.v1.Identity/GetPluginInfo
I0423 07:21:01.625305       1 utils.go:107] GRPC request:
I0423 07:21:01.625391       1 identityserver.go:32] Using default GetPluginInfo
I0423 07:21:01.625396       1 utils.go:112] GRPC response: name:"blobfuse.csi.azure.com" vendor_version:"v0.1.0-alpha"
I0423 07:21:01.626609       1 utils.go:106] GRPC call: /csi.v1.Identity/Probe
I0423 07:21:01.626619       1 utils.go:107] GRPC request:
I0423 07:21:01.626642       1 utils.go:112] GRPC response: ready:<value:true >
I0423 07:21:03.478535       1 controllerserver.go:98] begin to create container(pvc-fuse-dynamic-5a3b5ddd-6598-11e9-9ef8-000d3a0400a1) on account(fuse4f945a18659811e99ef) type(Standard_LRS) rg(mc_andy-aks1135_andy-aks1135_eastus2) location() size(10)
I0423 07:21:03.598393       1 controllerserver.go:119] create container pvc-fuse-dynamic-5a3b5ddd-6598-11e9-9ef8-000d3a0400a1 on storage account fuse4f945a18659811e99ef successfully
I0423 07:21:03.598410       1 utils.go:112] GRPC response: volume:<capacity_bytes:10737418240 volume_id:"mc_andy-aks1135_andy-aks1135_eastus2#fuse4f945a18659811e99ef#pvc-fuse-dynamic-5a3b5ddd-6598-11e9-9ef8-000d3a0400a1" volume_context:<key:"skuName" value:"Standard_LRS" > >
I0423 07:21:05.195464       1 utils.go:106] GRPC call: /csi.v1.Controller/CreateVolume
I0423 07:21:05.195499       1 utils.go:107] GRPC request: name:"pvc-4a92e455-6598-11e9-9989-0e3cd97702b7" capacity_range:<required_bytes:10737418240 > volume_capabilities:<mount:<fs_type:"ext4" mount_flags:"--file-cache-timeout-in-seconds=120" mount_flags:"--use-https=true" > access_mode:<mode:MULTI_NODE_MULTI_WRITER > > parameters:<key:"skuName" value:"Standard_LRS" >
I0423 07:21:05.238612       1 azure_storageaccount.go:103] found a matching account fuse4f945a18659811e99ef type Standard_LRS location eastus2
I0423 07:21:05.270246       1 controllerserver.go:98] begin to create container(pvc-fuse-dynamic-5b4cc2b5-6598-11e9-9ef8-000d3a0400a1) on account(fuse4f945a18659811e99ef) type(Standard_LRS) rg(mc_andy-aks1135_andy-aks1135_eastus2) location() size(10)
I0423 07:21:05.276393       1 controllerserver.go:119] create container pvc-fuse-dynamic-5b4cc2b5-6598-11e9-9ef8-000d3a0400a1 on storage account fuse4f945a18659811e99ef successfully
```

**fix solution**
 - option#1
1. search container name if `containerName` is empty
2. use a lock in `CreateVolume` func and release lock when quit `CreateVolume` func

 - option#2
1. Use a unique name for every CreateVolume request

**Workaround**
Use below storage class with `containerName` specified
```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: blobfuse.csi.azure.com
provisioner: blobfuse.csi.azure.com
parameters:
  skuName: Standard_LRS  #available values: Standard_LRS, Standard_GRS, Standard_RAGRS
  containerName: test #if container "test" does not exist, driver will create such container
reclaimPolicy: Retain #If set as "Delete" container would be removed after pvc deletion
volumeBindingMode: Immediate
```

BTW, also need to check whether azure file CSI driver has such issue.

**What you expected to happen**:

**How to reproduce it**:

**Anything else we need to know?**:

**Environment**:
- CSI Driver version:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/assign

**Comment by andyzhangx** [comment]
/assign @ZeroMagic 
Use volumename as container name would prevent this issue

**Comment by ZeroMagic** [comment]
ok. I will try to reproduce and fix it.

**Comment by andyzhangx** [comment]
already fixed by #44
